src/analysis.ts

```diff
@@ -110,3 +111,78 @@ export function intervalMatrix(scale: Scale) {
   }
   return result
 }
+
+// Interval matrix a.k.a the modes of a scale, but with step vectors.
+// Return the step vector for the j-step on mode i.
+export function stepVectorMatrix(scale: Scale): [[[[String, number]]]] {
+  console.log('stepVectorMatrix was called')
+  const result = []
+  const columns = [...Array(scale.size + 1).keys()]
+  for (let i = 0; i < scale.size; ++i) {
+    const mode = scale.rotate(i)
+    result.push(columns.map((j) => getStepVector(mode, j)))
+  }
+  return result
+}
+
+// Get j-step vector on tonic of `scale` and return it as an array of [String, number] pairs
+// (sorted in descending order by size).
+export function getStepVector(scale: Scale, intervalClass: number): [[String, number]] {
```

Comment:
Re-calculating the step vectors for each cell in the matrix is a waste of compute.

Reply:
Yeah, I need to somehow do this "statically wrt the `Scale`".